sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/nullExpressions.scala

```diff
@@ -91,29 +91,36 @@ case class Coalesce(children: Seq[Expression]) extends Expression {
     val codes = ctx.splitExpressionsWithCurrentInputs(
       expressions = evals,
       funcName = "coalesce",
-      makeSplitFunction = func =>
-        s"""
-           |do {
-           |  $func
-           |} while (false);
-         """.stripMargin,
-      foldFunctions = _.map { funcCall =>
-        s"""
-           |$funcCall;
-           |if (!${ev.isNull}) {
-           |  continue;
-           |}
-         """.stripMargin
-      }.mkString)
-
+      extraArguments = (ctx.javaType(dataType), ev.value) :: Nil,
```

Comment:
we don't need to pass it as a parameter, just create a local variable at the beginning of the generated method:
```
${ctx.javaType(dataType)} ${ev.value} = default...
do {
  $func
} while (false);
return ${ev.value};
```